tests/unit/lms/services/canvas_test.py

```diff
@@ -213,6 +213,16 @@ def test_find_matching_file_in_course_returns_the_matching_file_id(
         canvas_api_client.list_files.assert_called_once_with(sentinel.course_id)
         assert matching_file_id == str(sentinel.matching_file_id)
 
+    def test_fallback_to_guid_matching(self, finder, file_service):
```

Comment:
This test don't use the DB so setting up a `.return_value` with the GUID scenario is not testing much more.

Just asserting here that we are making the second type of query.